c10/util/complex_math.h

```diff
@@ -2,16 +2,16 @@ namespace std {
 
 // Exponential functions
 
-#if defined(CUDA_VERSION) && (CUDA_VERSION < 10000)
-#define CUDA92_BUG(x) thrust::complex<T>(x.real(), x.imag())
+#if defined(CUDA_VERSION) && (CUDA_VERSION < 10200)
+#define CUDA101_BUG(x) thrust::complex<T>(x.real(), x.imag())
 #else
-#define CUDA92_BUG(x) x
+#define CUDA101_BUG(x) x
 #endif
 
 template<typename T>
 C10_HOST_DEVICE inline c10::complex<T> exp(const c10::complex<T> &x) {
 #if defined(__CUDACC__) || defined(__HIPCC__)
-  return static_cast<c10::complex<T>>(thrust::exp(static_cast<thrust::complex<T>>(CUDA92_BUG(x))));
+  return static_cast<c10::complex<T>>(thrust::exp(static_cast<thrust::complex<T>>(CUDA101_BUG(x))));
```

Comment:
`CUDA_BUG` is not a descriptive enum name. 
Can we perhaps have a `template typename<T> inline thrust::complex<T> cast_to_thrust_complex(const c10::complex<T>&x)` that would be just a static cast for certain versions of thrust and explicit cast for others?
And than, line below would look something like:
```suggestion
  return static_cast<c10::complex<T>>(thrust::exp(cast_to_thrust_complex(x)));
```

Reply:
I restructured the PR

Reply:
Maybe `cast_to_thrust_complex_cuda101_bug`? It would be nice to make it clear that this is a workaround for a bug. So people won't be confused why the code now is looking so ugly, and in the future, when the bug is fixed, people know that it is ready to remove.